### PR TITLE
Feature/ember cli 3.1

### LINKED ISSUE #8: Update dependencies - remove deprecations
The following deprecation shows up due to old version of `ember-cli-version-checker`
```
DEPRECATION: An addon is trying to access project.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider doing: require("resolve").sync(something, { basedir: project.root }). Accessed from:   new NPMDependencyVersionChecker (<project_root>/node_modules/ember-bootstrap-changeset-validations/node_modules/ember-cli-version-checker/src/npm-dependency-version-checker.js:11:33)

```

### PR DESCRIPTION
**Pull request by nickschot**
fixes #8 

Fully updated to ember(-cli) 3.1. Also removes bower (dependencies). All applicable codemods have been applied.

There is still a linter error (due to new eslint-plugin-ember default configuration) of which I am currently unsure how I'd like to solve it.

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2016
+Copyright (c) 2018
```
Comment (simonihmig) [inline LICENSE.md]: I know this came automatically from ember-cli-update, but I think this should be left untouched.

File: addon/components/bs-form.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-  hasValidator: computed.notEmpty('model.validate'),
+  hasValidator: notEmpty('model.validate'),
 
   validate(model) {
     let m = model;
 
-    Ember.assert('Model must be a Changeset instance', m && typeof m.validate === 'function');
+    assert('Model must be a Changeset instance', m && typeof m.validate === 'function');
     return m.get('isValid') ? RSVP.resolve() : RSVP.reject();
   },
```
Comment (simonihmig) [inline addon/components/bs-form.js]: I think you can change `_initValidation` to just be an observer (i.e. remove the `on` wrapper`), and implement a plain init method like this to fix the ESLint issue:
```js
init() {
  this._super(...arguments);
  this. _initValidation();
}
```

File: addon/components/bs-form.js [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
 
-const { computed, RSVP, on, observer } = Ember;
-
 export default BsForm.extend({
-  
-  hasValidator: computed.notEmpty('model.validate'),
+  init() {
+    this._super(...arguments);
+
+    this. _initValidation();
```
[the remark this answers is not in the document]
Comment (nickschot) [reply addon/components/bs-form.js]: wow.....

[the remark this answers is not in the document]
Comment (nickschot) [reply addon/components/bs-form.js]: Should first finish the first coffee next time :')

**Comment by simonihmig** [comment]
@nickschot thanks again! Will release this shortly as 1.0.0! 🎉

**Comment by psteininger** [comment]
@simonihmig Thank you so much! I meant to get working on this next week myself. It's delightful to see it done :)

**Comment by simonihmig** [comment]
@psteininger all the credit for this update belongs to @nickschot! 😉

**Comment by psteininger** [comment]
in that case, thank you @nickschot !!

**Pull request merged**